build_tools/benchmarks/collect_compilation_statistics_test.py

```diff
@@ -30,54 +30,52 @@ def test_match_module_cmake_target_not_match(self):
   def test_parse_compilation_time_from_ninja_log(self):
     target1 = "benchmark_suites/TFLite/vmfb/deeplabv3.vmfb"
     target2 = "benchmark_suites/TFLite/vmfb/mobilessd.vmfb"
-    ninja_log = ("# ninja log v5\n"
-                 f"0\t100\taaa\tbuild/{target1}\taaa\n"
-                 f"130\t200\tbbb\tbuild/{target2}\tbbb\n")
+    ninja_log = StringIO("# ninja log v5\n"
+                         f"0\t100\taaa\tbuild/{target1}\taaa\n"
+                         f"130\t200\tbbb\tbuild/{target2}\tbbb\n")
 
-    with tempfile.NamedTemporaryFile("w") as f:
-      f.write(ninja_log)
-      f.flush()
-      target_map = parse_compilation_time_from_ninja_log(f.name)
+    target_map = parse_compilation_time_from_ninja_log(ninja_log)
 
     self.assertEqual(target_map, {target1: 100, target2: 70})
 
   def test_get_module_component_info(self):
-    with tempfile.NamedTemporaryFile() as module_file:
-      zip = zipfile.ZipFile(module_file, "w")
-      zip.writestr(VM_COMPONENT_NAME, b"abcd")
-      zip.writestr(CONST_COMPONENT_NAME, b"123")
-      zip.writestr("main_dispatch_0_vulkan_spirv_fb.fb", b"bindata1")
-      zip.writestr("main_dispatch_1_vulkan_spirv_fb.fb", b"bindata2")
-      zip.close()
-      module_file.flush()
-
-      component_sizes = get_module_component_info(module_file.name)
-      self.assertEqual(
-          component_sizes,
-          ModuleComponentSizes(file_size=os.stat(module_file.name).st_size,
-                               vm_component_size=4,
-                               const_component_size=3,
-                               total_dispatch_component_size=16))
+    module_file = BytesIO()
+    zip = zipfile.ZipFile(module_file, "w")
+    zip.writestr(VM_COMPONENT_NAME, b"abcd")
+    zip.writestr(CONST_COMPONENT_NAME, b"123")
+    zip.writestr("main_dispatch_0_vulkan_spirv_fb.fb", b"bindata1")
+    zip.writestr("main_dispatch_1_vulkan_spirv_fb.fb", b"bindata2")
+    zip.close()
```

Comment:
Use it as a context manager?

```suggestion
    with zipfile.ZipFile(module_file, "w") as zip:
      zip.writestr(VM_COMPONENT_NAME, b"abcd")
      zip.writestr(CONST_COMPONENT_NAME, b"123")
      zip.writestr("main_dispatch_0_vulkan_spirv_fb.fb", b"bindata1")
      zip.writestr("main_dispatch_1_vulkan_spirv_fb.fb", b"bindata2")
```

Reply:
Done.